### PR TITLE
Move off of dryice and to webpack for builds

### PR DESCRIPTION
**Pull request by fitzgen**
We were pretty much the last dryice user, and it isn't worth being on such a
minority build tool when we don't really want to maintain it
ourselves. Additionally, this lets us stop using amdefine and the AMD format,
which was an alright choice in 2011 but has increasingly shown its age. Regular
Common JS modules are used now, which should pave the way for better tree
shaking and unused submodule removal tooling. Note that in order to avoid
ruining `git blame`, this leaves each module indented by two spaces as it was
from the AMD `define` function, and wraps a block around them so that editors
don't freak out too much.

r? @jlongster 

cc @sokra

### CODE REVIEW

**Comment by fitzgen** [comment]
Looks like we will need to drop node 0.8.X for webpack which is fine by me. Its years old.

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+function run_test() {
+  for (var k in SOURCE_MAP_TEST_MODULE) {
+    if (/^test/.test(k)) {
+      SOURCE_MAP_TEST_MODULE[k](assert);
+    }
+  }
+}
+
+testFiles.forEach(function (file) {
+  module.exports.push({
```
Comment (jlongster) [inline webpack.config.js]: You're building each test file with webpack, which includes a `require` of the sourcemap lib. Won't that generate test files that each bundle in the entire sourcemap lib?

If you're generating files that are meant to run on our try server, you probably want to add an alias that resolves the sourcemap library to whatever the path is in firefox.

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    entry: path.join(testsDir, file),
+    output: {
+      path: path.join(distDir, "test"),
+      filename: file.replace(/\-/g, "_"),
+      library: "SOURCE_MAP_TEST_MODULE"
```
Comment (jlongster) [inline webpack.config.js]: what does this do? you don't have to have the `library` entry, you can just drop it if this is a dummy value

**Comment by jlongster** [comment]
Personally I'd say to just re-indent everything. You can pass `-w` to `git blame` to just ignore changes that are just whitespace, which you probably want to make the default anyway.

My main concern is building the test files. r+ if you make sure it's not actually bundling in the sourcemap library for all of them, which doesn't seem desirable

File: webpack.config.js (hunk 2 above)
Comment (jlongster) [inline webpack.config.js]: Oh, ignore this comment. I remember what this option does now. It exports the object and names it that.

File: webpack.config.js (hunk 1 above)
Comment (jlongster) [inline webpack.config.js]: Ok, I just looked at the test files and they `require` specific individual modules from the project. So you probably intended to just bundle them all together. Makes sense.

**Comment by jlongster** [comment]
I looked at it further and I understand what you're doing now. r+

**Comment by fitzgen** [comment]
Thanks for the review!

I'd rather not re-indent, even with `git blame -w` for the sake of ease of merging outstanding PRs and branches in the future.

File: webpack.config.js (hunk 1 above)
Comment (fitzgen) [inline webpack.config.js]: Yeah some duplication, but only in tests. The build process is just so much more sane than it used to be, that I think this is worth it.

**Comment by zertosh** [comment]
@fitzgen Can you publish this please?

**Comment by fitzgen** [comment]
I will later today, after @jlongster reviews the follow ups in #206.

**Comment by jlongster** [comment]
@fitzgen You can publish it (I only had 1 or 2 nits I think) but followup with looking into TravisCI for generating builds if you want later, doesn't need to block it.

**Comment by fitzgen** [comment]
Its been published.

**Pull request merged**